### ISSUE TITLE
Upgrade React Native

### ISSUE THREAD

**Issue by c3bryant** [UCSD/campus-mobile#220]
Upgrade the campus-mobile project to use the latest stable version we are able to achieve within the timeframe of the Abra milestone.

This upgrade will be incompatible with certain node modules which will also need to be updated.

This issue will be completed upon successful verification of all app functionality.

**Comment by Gulci** [comment]
c5703c4: Updated to React Native 0.44.0.

Most notably, a lot of things become deprecated after this point, including react.createClass(). Any components using middleware have to be updated or changed.

**Comment by Gulci** [comment]
2dae3c7 updates dependencies and fast forwards most of them to React Native 0.44.0 support.

Remaining todo: fix usage of View.propTypes, which has been deprecated in 0.44.0.

**Comment by Gulci** [comment]
4e5ff3f updates the project to React Native 0.44.3.

**Comment by Gulci** [comment]
e160d83 updates the project to React Native 0.51.0. Further updates are blocked by a bug in 0.54.0 that prevents react-native-maps from being included correctly. All dependencies have been updated!

**Comment by Gulci** [comment]
Dependent on #232.

**Comment by c3bryant** [comment]
Verified and merged into v5.6-rc
